### PR TITLE
feat(marxan-runner): pipe assets to input files

### PR DESCRIPTION
**Pull request by kgajowy**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts [hunk 1]
```diff
@@ -52,17 +81,35 @@ afterAll(async () => {
...
+  },
+  {
+    name: `puvsp.dat`,
+    assetUrl: `http://localhost:3000/puvsp.dat`,
+    targetRelativeDestination: `input/puvsp.dat`,
+  },
+  {
+    name: `puvsp_sporder.dat`,
+    assetUrl: `http://localhost:3000/puvsp_sporder.dat`,
+    targetRelativeDestination: `input/puvsp_sporder.dat`,
```
Comment (hotzevzl) [inline api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts:108]: `puvspr*` (missing the final `r`)

**Review by hotzevzl — APPROVED** [review]

File: api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts (hunk 1 above)
Comment (kgajowy) [reply api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts:108]: @hotzevzl 
image

could be but it was taken directly from marxan's example ; )

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/scenario-data/derive-scenario.facade.ts [hunk 2]
```diff
@@ -1,11 +1,48 @@
...
 
 @Injectable()
 export class DeriveScenarioFacade implements InputFiles {
-  async include(scenarioId: string, directory: string): Promise<void> {
-    // get all scenario data from db
-    // and transform them into files required by marxan
+  constructor(private readonly httpService: HttpService) {}
+
+  async include(directory: string, assets: Assets): Promise<void> {
+    // TODO security: ensure that none of the target filename starts with either `/` or `.` (tree traversing)
```
Comment (Dyostiq) [inline api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/scenario-data/derive-scenario.facade.ts:11]: https://nodejs.org/en/knowledge/file-system/security/introduction/

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/scenario-data/derive-scenario.facade.ts [hunk 3]
```diff
@@ -1,11 +1,48 @@
...
+        .get(sourceUri, {
+          responseType: 'stream',
+        })
+        .toPromise()
+        .then((response) => {
+          writer.write(response.data);
+          writer.end();
+        })
+        .catch(reject);
+    });
```
Comment (Dyostiq) [inline api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/scenario-data/derive-scenario.facade.ts:38]: hmm, so if `stream` does not work and it is not really streaming, why do you use `createWriteStream`? I'd suggest to just use `writeFile`

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/scenario-data/derive-scenario.facade.ts [hunk 4]
```diff
@@ -1,11 +1,48 @@
-import { Injectable } from '@nestjs/common';
-import { InputFiles } from '../../ports/input-files';
+import { resolve, dirname } from 'path';
+import { createWriteStream, promises } from 'fs';
```
Comment (Dyostiq) [inline api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/scenario-data/derive-scenario.facade.ts:2]:
```ts
import { createWriteStream, promises as fsPromises } from 'fs';
```

`promises.mkdir` is surprising a bit :)

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/ports/input-files.ts [hunk 5]
```diff
@@ -1,6 +1,19 @@
...
+  PUNAME: string;
+  SPECNAME: string;
+  PUVSPRNAME: string;
+  BOUNDNAME: string;
+}
+
+export type Assets = {
+  // asset: keyof Settings;
+  url: string;
+  relativeDestination: string; // this
```
Comment (Dyostiq) [inline api/apps/geoprocessing/src/marxan-sandboxed-runner/ports/input-files.ts:11]: I don't understand the comments 😅

File: api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts [hunk 6]
```diff
@@ -1,28 +1,35 @@
-import { INestApplication, Injectable } from '@nestjs/common';
+import { INestApplication } from '@nestjs/common';
 import { Test, TestingModule } from '@nestjs/testing';
-import { copySync } from 'fs-extra';
+import AxiosMockAdapter from 'axios-mock-adapter';
```
Comment (Dyostiq) [inline api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts:3]: I'd stick to use nock, it's library agnostic and a bit easier to use

**Review by Dyostiq — APPROVED** [review]

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/scenario-data/derive-scenario.facade.ts (hunk 3 above)
Comment (kgajowy) [reply api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/scenario-data/derive-scenario.facade.ts:38]: Thought maybe someone can pinpoint some obvious thing I missed. :(

File: api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts (hunk 1 above)
Comment (hotzevzl) [reply api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts:108]: @kgajowy which warez site did you download this copy of Marxan from? 😛 
they're listed as `puvspr.dat` and `puvspr_sporder.dat` in the manual - no big deal of course as we can name these files as we wish, but just noting this to keep to the defaults mentioned in the manual

Comment (kgajowy) [reply api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts:108]: @hotzevzl https://github.com/Marxan-source-code/marxan#test-data :)

image

🤷

File: api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts (hunk 6 above)
Comment (kgajowy) [reply api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts]: https://github.com/Vizzuality/marxan-cloud/pull/293/commits/1d75fdcb196091abd991263952816edf49b0d7f2

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/ports/input-files.ts (hunk 5 above)
Comment (kgajowy) [reply api/apps/geoprocessing/src/marxan-sandboxed-runner/ports/input-files.ts]: https://github.com/Vizzuality/marxan-cloud/pull/293/commits/1d75fdcb196091abd991263952816edf49b0d7f2

**Pull request merged**